presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -2615,8 +2615,27 @@ core::PlanFragment VeloxQueryPlanConverterBase::toVeloxQueryPlan(
                 toJsonString(systemPartitioningHandle->function));
         }
       }
+      case protocol::SystemPartitioning::SCALED: {
+        VELOX_USER_CHECK(
+            systemPartitioningHandle->function ==
+                protocol::SystemPartitionFunction::ROUND_ROBIN,
+            "Unsupported partitioning function: {}",
+            toJsonString(systemPartitioningHandle->function));
+        planFragment.planNode = std::make_shared<core::PartitionedOutputNode>(
+            "root",
+            partitioningKeys,
+            1,
+            core::PartitionedOutputNode::Kind::kArbitrary,
+            partitioningScheme.replicateNullsAndAny,
+            std::make_shared<RoundRobinPartitionFunctionSpec>(),
+            std::move(outputType),
+            std::move(sourceNode));
+        return planFragment;
+      }
       default:
-        VELOX_FAIL("Unsupported kind of SystemPartitioning");
+        VELOX_FAIL(
```

Comment:
```
s/VELOX_FAIL/VELOX_USER_FAIL/?
```

Reply:
I would leave it as it is because the SystemPartitioning is assigned by coordinator not user, so if a unsupported SystemPartitioning is passed in, it's a system error not user error